### ISSUE TITLE
[SDW] timeouts with RT711 (non-SDCA) registers - jack init disable issue?

### ISSUE THREAD

**Issue by plbossart** [thesofproject/linux#2629]
Intel internal daily tests 1118 shows the following errors

````
kernel: [10520.802203] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x40020000: GLB_PM_MSG: CTX_RESTORE
kernel: [10520.802228] sof-audio-pci 0000:00:1f.3: ipc tx: 0x40010000: GLB_PM_MSG: CTX_SAVE
kernel: [10520.802364] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x40010000: GLB_PM_MSG: CTX_SAVE
kernel: [10520.802482] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x20140000 successful
kernel: [10521.358781] intel-sdw intel-sdw.0: IO transfer timed out, cmd 3 device 1 addr 3501 len 1
kernel: [10521.358796] soundwire sdw-master-0: trf on Slave 1 failed:-110 write addr 3501 count 0
kernel: [10521.862761] intel-sdw intel-sdw.0: IO transfer timed out, cmd 3 device 1 addr 3819 len 1
kernel: [10521.862777] soundwire sdw-master-0: trf on Slave 1 failed:-110 write addr 3819 count 0
kernel: [10522.366733] intel-sdw intel-sdw.0: IO transfer timed out, cmd 3 device 1 addr 3821 len 1
kernel: [10522.366747] soundwire sdw-master-0: trf on Slave 1 failed:-110 write addr 3821 count 0
kernel: [10522.870708] intel-sdw intel-sdw.0: IO transfer timed out, cmd 3 device 1 addr 3855 len 1
kernel: [10522.870723] soundwire sdw-master-0: trf on Slave 1 failed:-110 write addr 3855 count 0
kernel: [10522.870734] rt711 sdw:0:25d:711:0: in rt711_jack_init disable
kernel: [10523.374683] intel-sdw intel-sdw.0: IO transfer timed out, cmd 3 device 1 addr 3501 len 1
kernel: [10523.374698] soundwire sdw-master-0: trf on Slave 1 failed:-110 write addr 3501 count 0
kernel: [10523.375638] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x9010f0f successful
kernel: [10523.375816] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f successful
kernel: [10523.375832] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 0 : core_mask 1
````

Unclear if it's related to the calibration issues? @bardliao @shumingfan  FYI

**Comment by shumingfan** [comment]
@plbossart Do you have the whole log? 
Is it the disabled jack detection case?

**Comment by plbossart** [comment]
Here's the log. that's for a kmod load/unload test on TGL RVP with the topic/sof-dev branch
cmd: TPLG=sof-tgl-rt711-rt1308-4ch.tplg ~/sof-test/test-case/check-kmod-load-unload.sh -l 50

[dmesg.txt](https://github.com/thesofproject/linux/files/5680168/dmesg.txt)

It doesn't contain the dynamic debug traces for rt711 so it's of limited use.
@bardliao @fredoh9 what does it take to add rt711 information to the sof-dyndbg.conf?

**Comment by shumingfan** [comment]
@plbossart I checked the dmesg.txt, but I'm not sure the root cause for now.
I suspect that the timing of the DSP core going to suspend mode and the jack init disable is too close.
Is it possible that the DSP core suspended/disabled already?

**Comment by shumingfan** [comment]
I feel this issue is the same as https://github.com/thesofproject/linux/issues/2627.

**Comment by plbossart** [comment]
@shumingfan this happens with the regular RT711, not the SDCA version. I updated the description to make it clearer.

**Comment by bardliao** [comment]
This is fixed by #2640
